src/checkServerVersion.ts

```diff
@@ -82,3 +123,120 @@ function serverVersionInfo(config: WorkspaceConfiguration): {
     configurationTarget,
   };
 }
+
+export function loadSnapshotVersions(): Promise<MetalsVersion[]> {
+  const url =
+    "https://oss.sonatype.org/content/repositories/snapshots/org/scalameta/metals_2.12/";
+  const ps = new Promise<string>((resolve, reject) => {
+    http.get(url, (resp) => {
+      let body = "";
+      resp.on("data", (chunk) => (body += chunk));
+      resp.on("end", () => resolve(body));
+      resp.on("error", (e) => reject(e));
+    });
+  });
+
+  return ps.then((text) => {
+    const root = parse(text);
+    const versions = root.getElementsByTagName("tr").flatMap((e) => {
+      const elem = e.getElementsByTagName("td")[0];
+      if (elem) {
+        const text = elem.text;
+        if (text.endsWith("/")) {
+          const parsed = parseMetalsVersion(text);
+          return parsed ? [parsed] : [];
+        } else {
+          return [];
+        }
+      } else {
+        return [];
+      }
+    });
```

Comment:
I wonder if we could add some marker class to `td` element and then extract it string using regex `<td class="marker">.*<\/td>`